File: solutions/uri/1030/1030.clj
(ns main)

(require '[clojure.string :as str])

(defn josephus
  [n, k]
  (if (= n 1)
    1
    (-> (recur (dec 1) k)
        (+ k)
        (- 1)
        (mod n)
        (+ 1))))

(defn main
  [c q]
  (let [numbers (map #(Integer/parseInt %) (str/split (read-line) #" "))]
    (println (format "Case %d: %d" c (josephus (get numbers 0) (get numbers 1))))
    (when (< c q)
      (recur (inc c) q))))

(let [q (Integer/parseInt (read-line))]
    (main 1 q))

File: solutions/uri/1030/1030.py
import sys

sys.setrecursionlimit(10002)


def josephus(n: int, k: int) -> int:
    if n == 1:
        return 1

    return ((josephus(n - 1, k) + k - 1) % n) + 1


_ = input()

i: int
line: str
for i, line in enumerate(sys.stdin):
    n: int
    k: int
    n, k = map(int, line.split())

    print(f'Case {i + 1}: {josephus(n, k)}')
